Makefile tries to access git inside docker

**Describe the bug**
The `build-pdk` rule in the openlane `Makefile` creates a docker instance to run the `Makefile` in `open_pdks`. The `open_pdks`'s `Makefile` calls git to download various prerequisites. On some installations (e.g. google cloud engine), the user ID that is set when docker is instantiated (`-u $(shell id -u $(USER)):$(shell id -g $(USER)) $(IMAGE_NAME)`) does not exist in the passwd file, which causes git to abort.

**To Reproduce**
Steps to reproduce the behavior:
1. On an installation where the user ID does not exist in the passwd file, run `make pdk`
2. If the process dies with the following
```
AttributeError: 'FileNotFoundError' object has no attribute 'read'
failed
CondaEnvException: Pip failed
```
Edit `$PDK_ROOT/skywater-pdk/environment.yml` to change `file:requirements.txt` to `requirements.txt`
and then
```
make skywater-library
make open_pdks 
make build-pdk 
```
3. You should see the following error.
```
Downloading alphanumeric library from https://github.com/PaulSchulz/sky130_pschulz_xx_hd
Cloning https://github.com/PaulSchulz/sky130_pschulz_xx_hd to ../sources/sky130_ml_xx_hd
Cloning into '../sources/sky130_ml_xx_hd'...
fatal: unable to look up current user in the passwd file: no such user
Unexpected end of command stream
make: *** [alpha-repo] Error 128
make: *** [build-pdk] Error 2
```

**Expected behavior**
Completion without errors.

**Desktop:**
 - OS: CentOS7 (Google cloud engine)
 - Openlane Version : v0.20
 - open_pdks Version: commit 6de165cc41b9f1050006d37288c8121c23756406
 - skywater-pdk Version: commit db2e06709dc3d876aa6b74a5f3893fa5f1bc2a6e
 
**Additional context**
This problem can be fixed by downloading the prerequisites between 2  docker instances in the openlane `Makefile`
 as follows:
```
 build-pdk: $(PDK_ROOT)/open_pdks $(PDK_ROOT)/skywater-pdk
          [ -d $(PDK_ROOT)/sky130A ] && \
                (echo "Warning: A sky130A build already exists under $(PDK_ROOT). It will be deleted first!" && \
                sleep 5 && \
                rm -rf $(PDK_ROOT)/sky130A) || \
                true
+       docker run --rm -v $(OPENLANE_DIR):/openLANE_flow -v $(PDK_ROOT):$(PDK_ROOT) -e PDK_ROOT=$(PDK_ROOT) -u $(shell id -u $(USER)):$(shell id -g $(USER)) $(IMAGE_NAME) sh -c " cd $(PDK_ROOT)/open_pdks && \
+               ./configure --enable-sky130-pdk=$(PDK_ROOT)/skywater-pdk/libraries --with-sky130-local-path=$(PDK_ROOT)"
+       cd $(PDK_ROOT)/open_pdks/sky130 && \
+               make veryclean && \
+               make prerequisites
         docker run --rm -v $(OPENLANE_DIR):/openLANE_flow -v $(PDK_ROOT):$(PDK_ROOT) -e PDK_ROOT=$(PDK_ROOT) -u $(shell id -u $(USER)):$(shell id -g $(USER)) $(IMAGE_NAME) sh -c " cd $(PDK_ROOT)/open_pdks && \
-               ./configure --enable-sky130-pdk=$(PDK_ROOT)/skywater-pdk/libraries --with-sky130-local-path=$(PDK_ROOT) && \
                cd sky130 && \
-               make veryclean && \
                make && \
                make install-local && \
                make clean" 
```
And since it is a makefile, the leading spaces in the above example must be changed to tabs.

--- a/Makefile
+++ b/Makefile
@@ -130,9 +130,11 @@ build-pdk: $(PDK_ROOT)/open_pdks $(PDK_ROOT)/skywater-pdk
 		rm -rf $(PDK_ROOT)/sky130A) || \
 		true
 	$(ENV_COMMAND) sh -c " cd $(PDK_ROOT)/open_pdks && \
-		./configure --enable-sky130-pdk=$(PDK_ROOT)/skywater-pdk/libraries --with-sky130-local-path=$(PDK_ROOT) && \
-		cd sky130 && \
-		make veryclean && \
+		./configure --enable-sky130-pdk=$(PDK_ROOT)/skywater-pdk/libraries --with-sky130-local-path=$(PDK_ROOT)"
+	cd $(PDK_ROOT)/open_pdks/sky130 && \
+		$(MAKE) veryclean && \
+		$(MAKE) prerequisites
+	$(ENV_COMMAND) sh -c " cd $(PDK_ROOT)/open_pdks/sky130 && \
 		make && \
 		make install-local && \
 		make clean"

